bbq/vm/value.go

```diff
@@ -19,83 +19,64 @@
 package vm
 
 import (
-	"github.com/onflow/atree"
-
 	"github.com/onflow/cadence/bbq"
+	"github.com/onflow/cadence/common"
 	"github.com/onflow/cadence/interpreter"
 )
 
-type Value interface {
-	isValue()
-	StaticType(StaticTypeContext) bbq.StaticType
-	Transfer(
-		transferContext TransferContext,
-		address atree.Address,
-		remove bool,
-		storable atree.Storable,
-	) Value
-	String() string
-}
-
-type StaticTypeContext = interpreter.ValueStaticTypeContext
-
-type StorageContext = interpreter.StorageContext
-
-// TODO: Eventually use/switch-to `interpreter.ValueTransferContext`
-type TransferContext interface {
-	StorageContext
-	ReferenceTracker
-}
-
-type MemberAccessibleValue interface {
-	// TODO: See whether `Config` parameter can be removed from the below functions.
-	// Currently it's unknown because `AccountCapabilityControllerValue` members
-	// are not yet implemented.
-
-	GetMember(config *Config, name string) Value
-	SetMember(config *Config, name string, value Value)
-}
-
-type ResourceKindedValue interface {
-	Value
-	// TODO:
-	//Destroy(interpreter *Interpreter, locationRange LocationRange)
-	//IsDestroyed() bool
-	//isInvalidatedResource(*Interpreter) bool
-	IsResourceKinded() bool
-}
-
-// ReferenceTrackedResourceKindedValue is a resource-kinded value
-// that must be tracked when a reference of it is taken.
-type ReferenceTrackedResourceKindedValue interface {
-	ResourceKindedValue
-	IsReferenceTrackedResourceKindedValue()
-	ValueID() atree.ValueID
-	IsStaleResource() bool
-}
-
-// IterableValue is a value which can be iterated over, e.g. with a for-loop
-type IterableValue interface {
-	Value
-	Iterator() ValueIterator
-}
-
-type ValueIterator interface {
-	Value
-	HasNext() bool
-	Next(config *Config) Value
-}
+type Value = interpreter.Value
+
+//type MemberAccessibleValue interface {
```

Comment:
Can this commented out code now be removed, or is it still (going to be) needed?

Reply:
Its removed in a follow-up PR https://github.com/onflow/cadence/pull/3828